Makefile

```diff
@@ -62,7 +62,7 @@ build: commands
 .PHONY: test
 test:
 	@echo "running unit tests"
-	go test ./... -cover $(RACE) -tags deadlock
+	go test ./... -cover $(RACE) -tags deadlock -short
```

Comment:
I do not agree with this, benchmark tests should be moved to use `go bench`
We should file a separate issue to adjust the build process to allow benchmarks to be created.

Until we have done that I think we need to run as is and not add the `-short` option

Reply:
Yes, this test looks more like a benchmark test, can I update it to be a benchmark test in this PR?

Reply:
You can do that, it does require a bit more change but I am OK with that if you want to take that on.
We don't need the benchmark in the makefile as a separate target just yet, we can run them manually.